### PR TITLE
update AragurDev repo

### PR DESCRIPTION
**Pull request by aarcwb**
Implemented functionalities below:

SEND:
audio ok
vcard ok
location ok
image ok
video ok
document ok

RECEIVE:
image ok
audio ok
video ok
document ok
vcard ok
location ok

Added in yowsup-cli
/document send
/location
/vcard

### CODE REVIEW

**Comment by aarcwb** [comment]
@AragurDEV I would be happy if it was added in Special Thanks.

**Comment by AragurDEV** [comment]
Please add the dependencies in the setup.py so the build can pass

**Comment by aarcwb** [comment]
Done. I added the dependencies but I am not sure if it was added in first pull request or I need to do new one.

**Comment by AragurDEV** [comment]
Builds are still failimg... Please fix

**Comment by aarcwb** [comment]
@AragurDEV I uninstall audioread and preview-generator and run 'sudo python3 setup.py install' with no erros.

Could give me mores details about the error?

**Comment by AragurDEV** [comment]
Just look at the build log on travis...

**Comment by jcerdan** [comment]
Hi,

Sorry for the man-in-the-middle comment.
I think this should be corrected:
`
resultGetPictureIqProtocolEntiy.writeToFile("/home/anderson/Desktop/%s_%s.jpg" % (getPictureIqProtocolEntity.getTo(), "preview" if resultGetPictureIqProtocolEntiy.isPreview() else "full"))
`
/home/anderson/Desktop/ should be /tmp/ ?

in file:
yowsup/demos/cli/layer.py , line 581

**Comment by AragurDEV** [comment]
First of all don't use paths at all. 
Due to different operating systems.
Find another way. You can use the running path as your temp dir.
Secondly the build is failing due to 'ERROR: Failure: KeyError ("Couldn't find field com.whatsapp.proto.ImageMessage.file_enc_sha256")'
Try to fix it. Maybe you have to edit the tests.
Also please come and hangout on our discord so we can look into it together

**Comment by aarcwb** [comment]
@jcerdan Thanks, I forget to remove this line. I updated my repo and now I am trying to solve the build error to pull request on @AragurDEV repo

**Comment by aarcwb** [comment]
@AragurDEV probably you are right about edit the tests because in my local environment the setup run successful and I checked pa_pb2.py file and file_enc_sha256 field exist.

How can i visualize the tests?

**Comment by AragurDEV** [comment]
@aarcwb there are special test files:
for example look for test_* in https://github.com/AragurDEV/yowsup/tree/master/yowsup/layers/protocol_media/protocolentities

**Comment by aarcwb** [comment]
@AragurDEV Done! I removed file_enc_sha256 field from image and video message and now travis build finish without errors. I made some tests to send and receive all types of media and everything still works.

**Comment by AragurDEV** [comment]
Okay so 'file_enc_sha256' is not required anymore?

**Comment by aarcwb** [comment]
No. 
At least for android phone I received image and video normally.

File: yowsup/common/http/warequest.py [hunk 1]
```diff
@@ -165,3 +173,39 @@ def sendRequest(host, port, path, headers, params, reqType="GET"):
...
+        url = 'https://'+host+path+"?"+params 
+        print(url)
+        return url 
+        
+    @staticmethod
+    def build_headers(headers_tuple):
+        headers_array = []
+        for idx in headers_tuple:
+            headers_array.append(idx + ":"+headers_tuple[idx])
+        print(headers_array)
```
Comment (mikesalmonuk) [inline yowsup/common/http/warequest.py]: Is it possible to remove debug code? I know this PR is already merged, but better without debug prints.

File: yowsup/common/http/warequest.py [hunk 2]
```diff
@@ -165,3 +173,39 @@ def sendRequest(host, port, path, headers, params, reqType="GET"):
...
+        c.perform()
+        c.close()
+        data = bytes_buffer.getvalue().decode('utf-8')
+        return json.loads(data)
+
+    @staticmethod
+    def build_get_url(host, path, params):
+        params = urlencode(params)
+        url = 'https://'+host+path+"?"+params 
+        print(url)
```
Comment (mikesalmonuk) [inline yowsup/common/http/warequest.py]: Is it possible to remove debug code? I know this PR is already merged, but better without debug prints.

File: yowsup/common/http/warequest.py (hunk 1 above)
Comment (AragurDEV) [reply yowsup/common/http/warequest.py]: Yes, i missed that.
Please create another pull request for this :)

Comment (mikesalmonuk) [reply yowsup/common/http/warequest.py]: Cool, will do. There was plenty of debug statements and prints here, so will try do a single PR for all of them!

**Pull request merged**